### PR TITLE
[PUBDEV-9101][DevOps] Jenkins pipeline for steam prisma scan

### PR DESCRIPTION
**Pull request by ChathurindaRanasinghe**
Jenkins Pipeline URL: http://mr-0xg1:8080/job/H2O-3/job/steam-prisma-scan/
Sample run: http://mr-0xg1:8080/job/H2O-3/job/steam-prisma-scan/57/

Notes: Enter the correct branch or tag in the input and run manually.

Parent Issues:

- https://github.com/h2oai/h2o-ops/issues/336
- https://github.com/h2oai/h2o-3/issues/15542

### CODE REVIEW

File: h2o-assemblies/steam/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        timestamps()
+    }
+
+    stages {
+        stage('0. Init'){
+            steps{
+                script{
+                    checkout(
+                        scm: [$class: 'GitSCM',
+                            branches: [[name: '${BRANCH_OR_TAG}'], [name: 'devops/steam-prisma-jenkins']],
```
Comment (hasithjp) [inline h2o-assemblies/steam/Jenkinsfile:27]: do we need both branches here?

File: h2o-assemblies/steam/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    project: '',
+                    resultsFile: 'prisma-cloud-scan-results.json',
+                    ignoreImageBuildTime:true
+                }
+                
+            }
+        }
+        stage('3. Export results to CSV'){
+            steps{
+                sh "curl -k -u ops:OPS@h2oai https://mr-0xz1:8083/api/v1/scans/download?search=${steamImage} > ${steamImage}.csv"
```
Comment (hasithjp) [inline h2o-assemblies/steam/Jenkinsfile:81]: Not a good practice harcode creds. please fetch through a Jenkins cred

File: h2o-assemblies/steam/Jenkinsfile (hunk 1 above)
Comment (ChathurindaRanasinghe) [reply h2o-assemblies/steam/Jenkinsfile:27]: Both branches were added to make sure that Dockerfiles exists irrespective of the input.

File: h2o-assemblies/steam/Jenkinsfile (hunk 2 above)
Comment (ChathurindaRanasinghe) [reply h2o-assemblies/steam/Jenkinsfile:81]: Sure

**Comment by mn-mikke** [comment]
@ChathurindaRanasinghe Thanks a lot for a great job! 

May I ask you to
- Move Docker files to a new dicrectory `docker/prisma`
- Move/Rename the jenkins file to `scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan`

File: scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            }
+        }
+        stage('2. Scan jar using Prisma'){
+            steps {
+                script{
+                    branchOrTag = "${BRANCH_OR_TAG}".replace('/','-')
+                    steamImage = "h2o-assemblies/steam:${BUILD_NUMBER}-${branchOrTag}"
+
+                    sh """
+                        git checkout devops/steam-prisma-jenkins
```
Comment (mn-mikke) [inline scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan:59]: Do we need this line? Is the `devops/steam-prisma-jenkins` going live forever?

Comment (ChathurindaRanasinghe) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan:59]: same here

File: scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            steps{
+                script{
+                    checkout(
+                        scm: [$class: 'GitSCM',
+                            branches: [[name: '${BRANCH_OR_TAG}'], [name: 'devops/steam-prisma-jenkins']],
+                            userRemoteConfigs: [[url: 'https://github.com/h2oai/h2o-3.git']]
+                        ]
+                    )
+
+                    sh "git checkout devops/steam-prisma-jenkins"
```
[the remark this answers is not in the document]
Comment (mn-mikke) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan:32]: Wouldn't the master branch serve better for this purpose? The master branch is always the most up to date one and never gets deleted. WDYT?

[the remark this answers is not in the document]
Comment (ChathurindaRanasinghe) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan:32]: yeah we can use master branch in that case. What I said was if the scans will run for new branches after merging with master the we would not need to checkout at all.

[the remark this answers is not in the document]
Comment (mn-mikke) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan:32]: I think we can support only future branches for now.

[the remark this answers is not in the document]
Comment (ChathurindaRanasinghe) [reply scripts/jenkins/jenkinsfiles/Jenkinsfile-PrismaScan]: done

**Review by mn-mikke — APPROVED** [review]
LGTM, thank you!

**Comment by mn-mikke** [comment]
@ChathurindaRanasinghe Please rebase your PR to current master to fix failing build.

**Comment by mn-mikke** [comment]
@hasithjp do you have any further comments or can we merge?

**Comment by ChathurindaRanasinghe** [comment]
> @ChathurindaRanasinghe is this pr to current master? I see also different changes in the PR.

I rebased the branch to the current master.

**Review by hasithjp — APPROVED** [review]
LGTM!

**Comment by mn-mikke** [comment]
Thanks a lot @ChathurindaRanasinghe, merging the PR

**Pull request merged**